src/core/pointcloud/qgscopcpointcloudindex.cpp

```diff
@@ -57,31 +59,36 @@ std::unique_ptr<QgsPointCloudIndex> QgsCopcPointCloudIndex::clone() const
   return std::unique_ptr<QgsPointCloudIndex>( clone );
 }
 
-void QgsCopcPointCloudIndex::load( const QString &fileName )
+void QgsCopcPointCloudIndex::load( const QString &urlString )
 {
-  mUri = fileName;
-  mCopcFile.open( QgsLazDecoder::toNativePath( fileName ), std::ios::binary );
-
-  if ( !mCopcFile.is_open() || !mCopcFile.good() )
+  QUrl url = urlString;
+  // Treat non-URLs as local files
+  if ( url.isValid() && ( url.scheme() == "http" || url.scheme() == "https" ) )
+    mAccessType = Remote;
+  else
   {
-    mError = tr( "Unable to open %1 for reading" ).arg( fileName );
-    mIsValid = false;
-    return;
+    mAccessType = Local;
+    mCopcFile.open( QgsLazDecoder::toNativePath( urlString ), std::ios::binary );
```

Comment:
It looks like the check for `mCopcFile.is_open()` and `mCopcFile.good()` got lost here, with early error reporting that the file can't be opened?